### PR TITLE
[DO NOT REVIEW] Don't use bucket pools

### PR DESCRIPTION
**Pull request by justinjc**

### CODE REVIEW

File: src/dbnode/storage/series/buffer.go [hunk 1]
```diff
@@ -930,12 +931,10 @@ func (b *BufferBucket) merge() (int, error) {
 		start   = b.start
 		readers = make([]xio.SegmentReader, 0, len(b.encoders)+len(b.bootstrapped))
 		streams = make([]xio.SegmentReader, 0, len(b.encoders))
-		iter    = b.opts.MultiReaderIteratorPool().Get()
```
Comment (richardartoul) [inline src/dbnode/storage/series/buffer.go]: Am I missing something or was this `iter` completely unused?

Comment (justinjc) [reply src/dbnode/storage/series/buffer.go]: Yeah you're correct. I've fixed it locally, but of course it doesn't fix the issue haha.

**Comment by schallert** [comment]
FYI @justinjc you might need to rebase this to pick up most recent BK settings, or close the PR and reopen. I saw the build was hanging for a few days due to targeting issues and cancelled it

**Comment by justinjc** [comment]
Thanks @schallert! This review was mostly to show a change and not to merge this.

**Pull request closed without merging**